### PR TITLE
SLT-464 – Hide and reveal the Website Header based on scrolling direction

### PR DESCRIPTION
**Pull request by LL782**
These changes introduce new Javascript for controlling the hide-and-reveal behaviour of the Website Header.

When scrolling up, if the header is out of view it will appear at the top of the viewport.
When scrolling down the header will disappear from view.
When near the top, the header will scroll with the rest of the document, like a normal page element.

This is all behind a feature toggle called `toggle--slt-464-new-js`

The toggle is switched off for now as we are planning to refine the animation of the header and the tolerance around scrolling before releasing this feature. Please review and if happy we can merge into master (keeping the feature toggle off)

### CODE REVIEW

**Review by Dan-Bird — APPROVED** [review]
This looks good to me

File: src/assets/custom/css/_sass/_website-header.scss [hunk 1]
```diff
@@ -1,12 +1,21 @@
 .website-header {
+  background-color: white;
   font-size: 13px;
   height: 74px;
-  padding-top: 5px;
   padding-bottom: 5px;
-  position: sticky;
-  top: 0;
+  padding-top: 5px;
+  background-color: white;
```
Comment (LL782) [inline src/assets/custom/css/_sass/_website-header.scss]:
```suggestion
```

File: src/assets/custom/js/websiteHeader.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  function handleScrollPosition(latestWindowPosition) {
+    var upwardMovement = previousWindowPosition > latestWindowPosition;
+
+    if (atTheTop(latestWindowPosition)) {
+      restoreNaturalPosition();
+      return;
+    }
+
+    if (startingPositionIsInView(latestWindowPosition)) {
+      hideStickyHeader();
```
Comment (LL782) [inline src/assets/custom/js/websiteHeader.js]: We should rename this...

**Review by Dan-Bird — APPROVED** [review]
# 👍 

Happy with the new changes and to get this in.

**Review by AurelijaZuba — APPROVED** [review]

**Review by cherryt — APPROVED** [review]

**Review by mattgraygithub — APPROVED** [review]
Looks good to me

**Pull request merged**